internal/state/provider_schema_test.go

```diff
@@ -439,12 +439,16 @@ func TestStateStore_ProviderSchema_legacyAddress_exactMatch(t *testing.T) {
 			expectedDescription, ps.Provider.Description.Value)
 	}
 
-	// Check that detail has "fixed" legacy namespace
```

Comment:
What's the reason behind this change?

Reply:
Never mind, found it on the other PR.

Reply:
Good question.

The idea _was_ that if user is still on 0.12 we'd pull schemas attributed to legacy addresses, which can still be used for completion, hover etc., but linking to docs for example is much trickier as the Registry doesn't have a stable URL for legacy addresses. This is why we'd have previously attempted to "fix" the namespace as linking to `hashicorp/aws` is possible, when linking to `-/aws` is not.

However this required modification of the schema in-place essentially, as we place the generated links inside the schema. This is no longer a viable option as we'd have to keep copying the schema on read.

https://github.com/hashicorp/terraform-schema/pull/39 has more radical solution to that problem

Basically we can keep displaying `-/aws 3.0.0` in the detail as that's still better than not displaying anything, but we omit the links entirely.